harness/determined/deploy/gcp/cli.py

```diff
@@ -55,7 +70,33 @@ def deploy_gcp(command: str, args: argparse.Namespace) -> None:
             det_configs[arg] = args_dict[arg]
 
     # Handle down subcommand.
-    if command == "down":
+    if command == "down" and args.cluster_id:
+        if not args.project_id:
+            err_str = (
+                "Error: Project Id not provided. Please provide both project id"
+                + " and cluster id to delete the cluster."
+            )
+            print(
+                colored(err_str, "red"),
+            )
+            sys.exit(1)
+
+        comm = [
+            "det",
+            "deploy",
+            "gcp",
+            "up",
+            "--cluster-id",
+            args.cluster_id,
+            "--project-id",
+            args.project_id,
+            "--dry-run",
+        ]
+        gcp.run_command(comm, env)
```

Comment:
would it be possible to use `gcp.dry_run` instead, so we stay in the same python process rather than spawn another command?

Reply:
Yes I can if I initialize my own det_configs with the defaults before calling it. The only forseeable issue I see with this is that if in the future we change the defaults or add/delete arguments this manually initialized config list would also need to be changed. But if we call it through the command line it will dynamically initialize the configs according to the current defaults. Let me know if you'd like me to go ahead with the change!

Reply:
I think it'd be nice to do. running the same code in a subprocess from itself may be confusing for the debugging if something goes wrong.

also, which defaults is that? why can't you use the same `det_configs` here for both dry-run and delete?

Reply:
This is what `det_configs` looks like in delete:
`{'func': <function handle_down at 0x10623a3a0>, 'no_preflight_checks': False, 'no_wait_for_master': False, 'image_repo_prefix': 'determinedai', '_command': 'd|eploy', '_subcommand': 'gcp', 'cluster_id': 'new-c-sgup-3', 'project_id': 'dai-dev-55', 'local_state_path': '/Users/salonigupta/determined', 'yes': False, '_subsubcommand': 'down'}`
And this is what det configs looks like in dryrun: 
`{'func': <function handle_up at 0x104ea3430>, 'no_preflight_checks': False, 'no_wait_for_master': False, 'image_repo_prefix': 'determinedai', '_command': 'd|eploy', '_subcommand': 'gcp', 'cluster_id': 'new-c-sgup-3', 'project_id': 'dai-dev-55', 'dry_run': True, 'network': 'det-default', 'filestore_address': '', 'no_filestore': False, 'det_version': '0.23.1-dev0', 'region': 'us-west1', 'disk_size': 200, 'disk_type': 'pd-standard', 'environment_image': 'det-environments-6eceaca', 'local_state_path': '/Users/salonigupta/determined', 'preemptible': False, 'operation_timeout_period': '5m', 'master_instance_type': 'n1-standard-2', 'compute_agent_instance_type': 'n1-standard-32', 'aux_agent_instance_type': 'n1-standard-4', 'db_password': 'postgres', 'max_aux_containers_per_agent': 100, 'max_idle_agent_period': '10m', 'max_agent_starting_period': '20m', 'port': 8080, 'gpu_type': 'nvidia-tesla-t4', 'gpu_num': 4, 'min_dynamic_agents': 0, 'max_dynamic_agents': 5, 'min_cpu_platform_master': 'Intel Skylake', 'min_cpu_platform_agent': 'Intel Broadwell', 'scheduler_type': 'fair_share', 'preemption_enabled': False, 'cpu_env_image': '', 'gpu_env_image': '', 'tf_state_gcs_bucket_name': 'dai-dev-55-determined-deploy', '_subsubcommand': 'up', 'labels': {}}`

It is because `det deploy gcp up` has many more optional parameters that have defaults as compared to down

Reply:
There may be a smart way to get the argparse parser from our CLI parser pipeline, and then we'd get the defaults without explicitly duplicating them in the code. maybe at least we could add a comment / TODO about it?

Reply:
Added a Todo and manually added the default configs to run gcp.dry_run for now